src/CopyUsers.js

```diff
@@ -0,0 +1,26 @@
+import AWS from 'aws-sdk'
+import config from 'config'
+import { configureAWS } from './configureAWS'
+
+export default class CopyUsers {
+  constructor(userList) {
+    AWS.config = configureAWS(AWS.config)
+    this.userListString = JSON.stringify(userList)
+    this.s3 = new AWS.S3()
+    this.objectKey = `${new Date().toISOString()}/${config.get('userPoolId')}.json`
```

Comment:
Store users in a JSON file named after the user pool in a subdirectory that is the datestamp of the operation.

Reply:
wouldn't we want the directory named for the userPool, and the filename being the dataestamp?  Possibly even preceded by "users", so 234zzv9wer9ads8f/usersTIMESTAMP.json  ?   then I'd have all the files for the same userpool together and just look for the last one.

Also ... would we want to delete older files?

Reply:
> wouldn't we want the directory named for the userPool, and the filename being the dataestamp?

one possible argument for organizing by date before pool id is that it makes it easier to see things in chronological order by directory structure and file name, using lexical sorting in either `tree` or `ls`.  which would make it easy to see at a glance where the latest backup file was, without having to remember what order the pool IDs were used in.

>  Possibly even preceded by "users", so 234zzv9wer9ads8f/usersTIMESTAMP.json

i do like the idea of a filename that's more descriptive than just a pool id or a timestamp, and i do like having files that sort well as siblings even when their parent directory structures get flattened.  but i also like the idea of timestamp preceding pool id.  so my personal pref would be something like `DATE/user-backup_DATE_POOLID.json`, for e.g. a filename like `2019-05-01/user-backup_2019-05-01_234zzv9wer9ads8f.json`.

> Also ... would we want to delete older files?

i'm not sure if this is captured in a ticket at the moment, but my understanding from slack discussion yesterday was that we'd punt on cleanup till later.  i think @mjgiarlo's back of the envelope calculation expected < 25 MB of storage use per year if we don't purge anything, which seems pretty minimal.

Reply:
@ndushay :speech_balloon: 
> wouldn't we want the directory named for the userPool, and the filename being the datestamp?

Perhaps. But AFAIK we only have one user pool per env, so the top-level user pool dir is only so valuable in an env that won't have multiple pools.

> Possibly even preceded by "users", so 234zzv9wer9ads8f/usersTIMESTAMP.json ? then I'd have all the files for the same userpool together and just look for the last one.

Hmmm, how do you think the `users` prefix helps?

> Also ... would we want to delete older files?

I defer to @jermnelson. Jeremy, how much do we care about cleaning up our user pool backups in M3 (or even in M4)? It's fairly cheap to keep the files hanging around, and we can always clean later. I'm inclined to keep that out of this PR, but I'm happy to write up a new issue if this is something we care about!

Reply:
I don't think we need to delete the files at least for a couple of months into the project. To me this would be ongoing maintenance or part of the next work-cycle if we wanted to add a programmatic way to remove the old files.

Reply:
@jermnelson OK. I'll write up a ticket and it'll sit in the backlog.

Reply:
https://github.com/LD4P/sinopia_user_backup/issues/3